Content.Server/Spawners/EntitySystems/GenericSpawnerSystem.cs

```diff
@@ -73,31 +73,32 @@ private void Spawn(EntityUid uid, GenericSpawnerComponent component)
         if (component.EntityTables.Count == 0)
             return;
 
-        foreach (var pair in component.EntityTables)
-        {
-            if (!_proto.TryIndex(pair.Key, out var entTable))
-            {
-                Log.Warning($"Referenced entity table prototype does not exist! Entity: {ToPrettyString(uid)}");
-                continue;
-            }
 
-            if (pair.Value != 1.0f && !_robustRandom.Prob(pair.Value))
-                continue;
+        if (component.Rolls is < 1 or > 100)
+        {
+            Log.Warning($"Invalid amount of rolls on entity table, value should be between 1 and 100. Entity: {ToPrettyString(uid)}");
+            return;
+        }
 
-            if (entTable.Weights.Count == 0)
+        foreach (var _ in Enumerable.Repeat(1, component.Rolls))
+        {
+            foreach (var pair in component.EntityTables)
```

Comment:
While the logic is a bit better, you'd still have to add a `break` once the entity is spawned to exit this inner foreach loop so it doesn't try to pick another table.

I'd also try to arrange the EntityTables in the yml so that the tables with smaller weights are listed first, so that you get the rare spawns more often if you intend to stick with probabilities.

To explain why, let's hypothetically imagine that the order we have is common -> common -> rare. Each common table has an 80% probability listed, and then the rare is a 5%. The first table fails 20% of the time before giving the second table a chance. The second table fails 20% of the time before giving the third table a chance.

The odds to get that 3rd table would be 20% (fail) * 20% (fail) * 5% (succeed) = 0.2%. Much more rare than we seemingly intended at 5% all because of the compounding probabilities. This is the kind of problem that arises when we roll for probability for each time individually rather than just picking it all at once with something like `Pick()` against a Dict of weights.

The reason why I preferred having the Dict be `<Table, Weights>` rather than `<Table, %>` is because you can just run the Pick function once and you'd know it'd pick a table pretty fairly. With probabilities and rolling one at a time, you can't really deal with the compounding probabilities.

Reply:
After double-checking the current code, it does indeed roll the rare probability first